app/helpers/audits_helper.rb

```diff
@@ -106,6 +106,41 @@ def audit_time(audit)
     date_time_absolute(audit.created_at)
   end
 
+  def audit_affected_locations(audit)
+    all_taxonomies = affected_taxonomies(audit)
+    return all_taxonomies if all_taxonomies.present?
+
+    base = audit.locations.authorized(:view_locations)
+    return _('none') if base.empty?
+
+    authorizer = Authorizer.new(User.current, base)
+    base.map do |location|
+      link_to_if_authorized location.name, hash_for_edit_location_path(location).merge(:auth_object => location, :permission => 'edit_locations', :authorizer => authorizer)
+    end.to_sentence.html_safe
+  end
+
+  def audit_affected_organizations(audit)
+    all_taxonomies = affected_taxonomies(audit)
+    return all_taxonomies if all_taxonomies.present?
+
+    base = audit.organizations.authorized(:view_organizations)
+    return _('none') if base.empty?
+
+    authorizer = Authorizer.new(User.current, base)
+    base.map do |organization|
+      link_to_if_authorized organization.name, hash_for_edit_organization_path(organization).merge(:auth_object => organization, :permission => 'edit_organizations', :authorizer => authorizer)
+    end.to_sentence.html_safe
+  end
+
+  def affected_taxonomies(audit)
+    begin
+      return _('all') unless audit.auditable_type.constantize.included_modules.include?(Taxonomix)
```

Comment:
This will return `all` for hosts and taxonomies, since they don't include `Taxonomix`, and might be true for other resources as well that have different ways of linking to taxonomies